filebeat/crawler/prospector_log.go

```diff
@@ -44,18 +44,21 @@ func (p *ProspectorLog) Init() {
 			// Make sure new harvester is started for all states
 			state.Finished = true
 			// Prospector must update all states as it has to detect also file rotation
-			p.Prospector.updateState(state)
+			p.Prospector.states.Update(state)
 		}
 	}
 
-	logp.Info("Previous states loaded: %v", len(p.Prospector.harvesterStates))
+	logp.Info("Previous states loaded: %v", p.Prospector.states.Count())
 }
 
 func (p *ProspectorLog) Run() {
 	logp.Debug("prospector", "Start next scan")
 
 	p.scan()
-	p.Prospector.cleanupStates()
+	// Only cleanup states if enabled
+	if p.config.IgnoreOlder != 0 {
+		p.Prospector.states.Cleanup(p.config.IgnoreOlder)
```

Comment:
Yes, this PR does not change the behaviour. For changing it see https://github.com/elastic/beats/issues/1600